### ISSUE TITLE
Cannot Set Password

### ISSUE THREAD

**Issue by ghdulrich** [agower/GeneHive#5]
I cannot get the password in there to save my life. I keep getting 401 errors. I see via debugging the API that the username is getting passed to the API but the password is not. I have my .netrc set up correctly (at least I think I do). This is happening on both linux and macos.

**Comment by agower** [comment]
I'm not quite sure what the context of this issue is.  Are you attempting to change the existing password for a user using the function `changePassword()`, or just trying to authenticate?

It sounds like you probably have your `~/.netrc` file set up correctly, but just to check, it should look like this:
```
machine operend.bu.edu login joeUser password myPa55w0rd
```
where any whitespace can be used as a delimiter.

**Comment by ghdulrich** [comment]
I didn't try to changePassword() I was just doing a listUsers() with my existing password set in my .netrc file. Is the .netrc file the only way to use a password? Is the code counting on curl to add the username & password via the .netrc file or does the code explicitly set the username:password in the header? If so, I'll test curl and see how that behaves. One thing to note here, we may be better off using tokens with some kind of secret mechanism that I'm sure R must provide now that access tokens are implemented on the API side

**Comment by agower** [comment]
> Is the .netrc file the only way to use a password?

Yes, I decided on .netrc as the only way to establish a connection, for a few reasons as outlined in the comments of the `hiveConnection()` function (starting at line 94 of [classes.R](https://github.com/agower/GeneHive/blob/master/R/classes.R)).

> Is the code counting on curl to add the username & password via the .netrc file or does the code explicitly set the username:password in the header?

The former; the `hiveConnection` object is a wrapper for a `CURLHandle`, an opaque, C-level structure implemented in the RCurl package, which contains a number of fields, including the username and password.  The actual API call is made through a call to `RCurl::curlPerform()` in the [httpRequest()](https://github.com/agower/GeneHive/blob/master/R/httpRequest.R) function.

> One thing to note here, we may be better off using tokens with some kind of secret mechanism that I'm sure R must provide now that access tokens are implemented on the API side

I totally agree.  I haven't looked into what kinds of data structures might be best for this but it's on my to-do list.

**Comment by agower** [comment]
Let me know if a Zoom call would be helpful to troubleshoot this together.

**Comment by ghdulrich** [comment]
How about we just let this issue  fade away assuming you'll be updating to tokens soon. Just FYI -- I've tried this on about 4 different computers and so far scc4 is the only one that works. None of the other installs will send the password from the .netrc file. Hmm
